### PR TITLE
Add script to create the NuGet package from the online build artifacts

### LINKED ISSUE #22: Automate the building of the NuGet package - Step 1
As discussed in https://github.com/libgit2/libgit2sharp/pull/1197 this is a first attempt at retrieving all the build artifacts.

/cc @bording

### PR DESCRIPTION
**Pull request by nulltoken**
Fix #22

PowerShell not being my forte, I'd be really grateful for an early feedback

/cc @bording @dahlbyk

### CODE REVIEW

File: download.build.artifacts.ps1 [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  $output = Invoke-RestMethod @irmParams -ContentType "application/json" -Method "Get" -Uri "$url"
+
+  if ($verbose) {
+    Write-Host -ForegroundColor "Gray" "output = $(ConvertTo-Json $output)"
+  }
+
+  return $output
+}
+
+function Extract-BuildIdentifier($statutes, $forContext) {
```
Comment (nulltoken) [inline download.build.artifacts.ps1]: I can't help but think this whole function is written in a very clumsy way... Any refactoring proposal?

File: download.build.artifacts.and.package.ps1 [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  $output = Invoke-RestMethod @irmParams -ContentType "application/json" -Method "Get" -Uri "$url"
+
+  if ($verbose) {
+    Write-Host -ForegroundColor "Gray" "output = $(ConvertTo-Json $output)"
+  }
+
+  return $output
+}
+
+function Extract-BuildIdentifier($statutes, $forContext) {
```
Comment (nulltoken) [inline download.build.artifacts.and.package.ps1]: Is there a way to make this more powershell-y?

**Comment by nulltoken** [comment]
Ok. I'm done for the first pass. This downloads the build artifacts and create the final NuGet packaged to be pushed.

This is (still) very crude but does the job (to my best knowledge).

Thoughts? 

/cc @ethomson @carlosmn

File: download.build.artifacts.and.package.ps1 (hunk 2 above)
Comment (dahlbyk) [inline download.build.artifacts.and.package.ps1]: 1. Typo in variable name
2. `$status = $statuses | where { $_.context -eq $forContext } | select -First 1`?

Comment (nulltoken) [inline download.build.artifacts.and.package.ps1]: @dahlbyk Brilliant! :heart: fixed

**Comment by bording** [comment]
I know you just merged this in, but I just wanted to say that I finally had some time to check it out and it looks good to me!

The only thing that you could do is add `-NoPackageAnalysis` to the `nuget.exe` call to avoid it displaying the "Assembly outside lib folder" warnings.

**Comment by nulltoken** [comment]
@bording Thanks for the feedback :heart: 

#27 should fix this.

**Pull request merged**